extractor/src/test/java/org/schabi/newpipe/extractor/services/peertube/PeertubeCommentsExtractorTest.java

```diff
@@ -18,74 +18,100 @@
+            for (final CommentsInfoItem c : comments.getItems()) {
+                assertFalse(Utils.isBlank(c.getUploaderUrl()));
+                assertFalse(Utils.isBlank(c.getUploaderName()));
+                assertFalse(Utils.isBlank(c.getUploaderAvatarUrl()));
+                assertFalse(Utils.isBlank(c.getCommentId()));
+                assertFalse(Utils.isBlank(c.getCommentText()));
+                assertFalse(Utils.isBlank(c.getName()));
+                assertFalse(Utils.isBlank(c.getTextualUploadDate()));
+                assertFalse(Utils.isBlank(c.getThumbnailUrl()));
+                assertFalse(Utils.isBlank(c.getUrl()));
+                assertFalse(c.getLikeCount() != -1);
+            }
+        }
 
-    @Test
-    public void testGetCommentsAllData() throws IOException, ExtractionException {
-        InfoItemsPage<CommentsInfoItem> comments = extractor.getInitialPage();
-        for (CommentsInfoItem c : comments.getItems()) {
-            assertFalse(Utils.isBlank(c.getUploaderUrl()));
-            assertFalse(Utils.isBlank(c.getUploaderName()));
-            assertFalse(Utils.isBlank(c.getUploaderAvatarUrl()));
-            assertFalse(Utils.isBlank(c.getCommentId()));
-            assertFalse(Utils.isBlank(c.getCommentText()));
-            assertFalse(Utils.isBlank(c.getName()));
-            assertFalse(Utils.isBlank(c.getTextualUploadDate()));
-            assertFalse(Utils.isBlank(c.getThumbnailUrl()));
-            assertFalse(Utils.isBlank(c.getUrl()));
-            assertFalse(c.getLikeCount() != -1);
+        private boolean findInComments(final InfoItemsPage<CommentsInfoItem> comments, final String comment) {
+            return findInComments(comments.getItems(), comment);
         }
-    }
 
-    private boolean findInComments(InfoItemsPage<CommentsInfoItem> comments, String comment) {
-        return findInComments(comments.getItems(), comment);
+        private boolean findInComments(final List<CommentsInfoItem> comments, final String comment) {
+            for (final CommentsInfoItem c : comments) {
+                if (c.getCommentText().contains(comment)) {
+                    return true;
+                }
+            }
+            return false;
+        }
     }
 
-    private boolean findInComments(List<CommentsInfoItem> comments, String comment) {
-        for (CommentsInfoItem c : comments) {
-            if (c.getCommentText().contains(comment)) {
-                return true;
-            }
+    public static class DeletedComments {
+        private static PeertubeCommentsExtractor extractor;
+
+        @BeforeClass
+        public static void setUp() throws Exception {
+            NewPipe.init(DownloaderTestImpl.getInstance());
+            extractor = (PeertubeCommentsExtractor) PeerTube
+                    .getCommentsExtractor("https://framatube.org/videos/watch/217eefeb-883d-45be-b7fc-a788ad8507d3");
+        }
+
```

Comment:
instead of checking whether there is no error (which will be broken for an unrelated error), you should check whether comment with id `9214` is not in the list of comment `id`s, via `PeertubeCommentsInfoItemExtractor#getCommentId()`

Reply:
>instead of checking whether there is no error (which will be broken for an unrelated error), you should check whether comment with id 9214 is not in the list of comment ids, via PeertubeCommentsInfoItemExtractor#getCommentId()

It's still a good idea to check that there are no errors in general.